### PR TITLE
NoJira: Current Scheme Status changes and phase progression

### PR DESCRIPTION
**Pull request by andrewrembrandt**
Scheme Status (Evaluation result and Withdrawal)
Update immediately when a scheme evaluation result is updated or is withdrawn from (numerical tests, sift, fsac, fsb)
TestDataGenerator updates to include matching statii

Have included a test demonstrating the correct progression updates to cumulative status

### CODE REVIEW

File: app/model/EvaluationResults.scala [hunk 1]
```diff
@@ -21,15 +21,40 @@ import model.persisted.SchemeEvaluationResult
...
+      case Red => Red
+      case Amber => Amber
+      case Withdrawn => Withdrawn
+    }
   }
   case object Amber extends Result {
     def toReportReadableString: String = "Amber"
+
+    def +(that: Result): Result = that match {
+      case Green => this
```
Comment (kjharkin) [inline app/model/EvaluationResults.scala]: May need to discuss this.

File: it/repositories/assessmentcentre/AssessmentCentreRepositorySpec.scala [hunk 2]
```diff
@@ -87,18 +103,21 @@ class AssessmentCentreRepositorySpec extends MongoRepositorySpec with ScalaFutur
     "update cumulative evaluation results from sift and phase 3" in {
```
Comment (kjharkin) [inline it/repositories/assessmentcentre/AssessmentCentreRepositorySpec.scala]: name

File: it/repositories/assessmentcentre/AssessmentCentreRepositorySpec.scala [hunk 3]
```diff
@@ -87,18 +103,21 @@ class AssessmentCentreRepositorySpec extends MongoRepositorySpec with ScalaFutur
...
       val nextResults = repository.nextApplicationForAssessmentCentre(1).futureValue
       nextResults mustBe List(
-        ApplicationForFsac("appId11",
-          PassmarkEvaluation("", Some(""), List(SchemeEvaluationResult(Finance, EvaluationResults.Green.toString),
-            SchemeEvaluationResult(Generalist, EvaluationResults.Red.toString)), "", Some("")),
-          List(SchemeEvaluationResult(Finance, EvaluationResults.Green.toString)))
+        ApplicationForFsac("appId11", ApplicationStatus.SIFT,
+          List(SchemeEvaluationResult(Finance, EvaluationResults.Green.toString),
+            SchemeEvaluationResult(Generalist, EvaluationResults.Red.toString),
+            SchemeEvaluationResult(DiplomaticService, EvaluationResults.Green.toString))
```
Comment (kjharkin) [inline it/repositories/assessmentcentre/AssessmentCentreRepositorySpec.scala]: This result looks incorrect.

Comment (benjstephenson) [inline it/repositories/assessmentcentre/AssessmentCentreRepositorySpec.scala]: updated test

**Pull request merged**